src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/AcspMembershipController.java

```diff
@@ -109,7 +110,9 @@ public ResponseEntity<Void> updateAcspMembershipForAcspAndId( final String xRequ
                     return new NotFoundRuntimeException( StaticPropertyUtil.APPLICATION_NAMESPACE, PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN );
                 } );
 
-        if ( UserRoleEnum.OWNER.getValue().equals( membershipIdAssociation.getUserRole() ) && acspMembershipService.fetchNumberOfActiveOwners( membershipIdAssociation.getAcspNumber() ) <= 1 ){
+        final var targetAcsp = acspProfileService.fetchAcspProfile( membershipIdAssociation.getAcspNumber() );
+        final var isLastOwner = UserRoleEnum.OWNER.getValue().equals( membershipIdAssociation.getUserRole() ) && acspMembershipService.fetchNumberOfActiveOwners( membershipIdAssociation.getAcspNumber() ) <= 1;
+        if ( isLastOwner && ( isOAuth2Request() || !( !isOAuth2Request() && targetAcsp.getStatus().equals( CEASED ) ) ) ){
```

Comment:
can we use ENUM for ceased or constant?
instead of this if can we just use
if(isLstOwner && status!=ceased) shoulnt this do the job?

Reply:
I think we are already using the enum for ceased.

Currently the if statement has this logic:
- OAuth2 can never remove the last owner
- API Key can only remove the last owner if the status is ceased
If we reduce the if statement to only check for the last owner and status ceased then, we aren't differentiating between OAuth2 and API Key, so both will be able to remove the last owner if the status is ceased.

Do we want to enable it for OAuth2 as well?